gui/src/app/pages/HomePage/SaveProjectWindow.tsx

```diff
@@ -211,4 +228,60 @@ const makeSPShareableLinkFromGistUrl = (gistUrl: string) => {
   return url;
 };
 
+type SaveToBrowserViewProps = {
+  fileManifest: Partial<FileRegistry>;
+  title: string;
+  onClose: () => void;
+};
+
+const SaveToBrowserView: FunctionComponent<SaveToBrowserViewProps> = ({
+  fileManifest,
+  title,
+  onClose,
+}) => {
+  // use IndexedDB to save the project
+  // https://developer.mozilla.org/en-US/docs/Web/API/IndexedDB_API/Using_IndexedDB
+
+  const handleSave = useCallback(async () => {
+    try {
+      const bpi = new BrowserProjectsInterface();
+      const existingProject = await bpi.loadProject(title);
+      if (existingProject) {
+        const overwrite = window.confirm(
+          `A project with the title "${title}" already exists. Do you want to overwrite it?`,
+        );
+        if (!overwrite) {
+          return;
+        }
+      }
+      await bpi.saveProject(title, fileManifest);
+    } catch (err: any) {
+      alert(`Error saving to browser: ${err.message}`);
+    }
+    onClose();
+  }, [title, fileManifest, onClose]);
+
+  return (
+    <div className="SaveToBrowserView">
+      <h3>Save to Browser</h3>
+      <p>
+        This project will be saved to your browser as &quot;{title}&quot;.&nbsp;
+        It will be available to you on this device, but not on other devices or
+        browsers.
```

Comment:
It should also probably be noted here that certain actions like clearing your browsers cache may lose this.

Reply:
I updated this to read

```
This project will be saved to your browser as &quot;{title}&quot;.&nbsp;
        It will be available to you on this device until you clear your browser
        cache, but not on other devices or browsers.
```